ui/appui-react/src/appui-react/layout/state/PanelState.ts

```diff
@@ -13,15 +13,13 @@ import type {
 } from "../widget-panels/PanelTypes";
 import { isHorizontalPanelSide } from "../widget-panels/Panel";
 import type { WidgetState } from "./WidgetState";
-
-/** @internal */
-export type PanelMaxSizeState = number | { readonly percentage: number };
+import type { StagePanelSizeSpec } from "../../stagepanels/StagePanelConfig";
 
 /** @internal */
 export interface PanelState {
   readonly collapseOffset: number;
   readonly collapsed: boolean;
-  readonly maxSize: PanelMaxSizeState;
+  readonly maxSize: StagePanelSizeSpec;
   readonly minSize: number;
```

Comment:
Why are we keeping `minSize`&`size` as a number while `maxSize` is a spec?

Reply:
`minSize` and `size` are used directly in components for calculations or defining styling e.g. in `Panel.tsx`, `Grip.tsx`. So it makes sense to keep them as pixel values (unless we want to do more refactoring). To have these properties as pixels, `NineZoneStateReducer` converts any percentages to pixels before setting `minSize` and `size` values for PanelState. That means `minSize` and `size` will always be pixels so having these properties as `StagePanelSizeSpec` does not make sense because they will never be set as percentage values (and when we access them in components mentioned above, we would have to write `panelState.size as number` which unneccesarily complicates the code).

Reply:
I see, we can leave it as is then.